src/frontend/js/components/Search/index.tsx

```diff
@@ -78,49 +78,69 @@ const Search = ({ context }: CommonDataProps) => {
     }
   }, [courseSearchResponse]);
 
+  // this is boilerplaty code to help generate a class after the .search__filters css transition ended
+  const [filtersPaneClosing, setFiltersPaneClosing] = useState(false);
+  const cssFiltersPaneTransitionDuration = 500;
+  useEffect(() => {
+    if (filtersPaneClosing) {
+      setTimeout(() => {
+        setFiltersPaneClosing(false);
+      }, cssFiltersPaneTransitionDuration);
+    }
+  }, [filtersPaneClosing]);
+
+  const onFiltersPaneCloseButtonClick = () => {
+    if (showFilters) {
+      setFiltersPaneClosing(showFilters);
+    }
+    setShowFilters(!showFilters);
+  };
+
```

Comment:
I must confess I'm not a huge fan of all this boilerplaty, sensitive code (the `cssFiltersPaneTransitionDuration` must match the value in the stylesheet). It's done to prevent having a UI jump when hiding the mobile filters pane.

This shows my goal, don't wanna start potentially spending time on something more elaborate without your opinion first @jbpenrath

Reply:
Instead of a timeout, you could use a event listener.

```ts
filterPaneRef.current?.addEventListener(
        'animationend',
        () => {
          setFiltersPaneClosing(false);
        },
        { once: true },
      );
```